### PR TITLE
Tasks-51729 ~ 51730 | 51731 ~ 51732: Add Entity of ChallengeEntity and AnnouncementEntity and add Model Challenge & Announcement | add EntityMapper

### PR DESCRIPTION
**Pull request by Rdidmohamedelhadi**

### CODE REVIEW

File: challenges-services/src/main/resources/db/changelog/challenge-rdbms.db.changelog-1.0.0.xml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    <!-- Managing both DB that use sequences and db that use auto increment -->
+    <property name="autoIncrement" value="true" dbms="mysql,mssql,h2,sybase,db2,hsqldb"/>
+    <property name="autoIncrement" value="false" dbms="oracle,postgresql"/>
+
+    <!-- Managing auto generation of timestamp by Database -->
+    <property name="now" value="now()" dbms="mysql,hsqldb,postgresql,h2"/>
+    <property name="now" value="sysdate" dbms="oracle"/>
+    <property name="now" value="CURRENT_TIMESTAMP" dbms="mssql"/>
+
+    <changeSet author="agenda" id="1.0.0-0" dbms="oracle,postgresql">
```
Comment (ahamdi) [inline challenges-services/src/main/resources/db/changelog/challenge-rdbms.db.changelog-1.0.0.xml]:
```suggestion
    <changeSet author="challenge" id="1.0.0-0" dbms="oracle,postgresql">
```

File: challenges-services/src/main/resources/db/changelog/challenge-rdbms.db.changelog-1.0.0.xml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        <createSequence sequenceName="SEQ_CHALLENGE_ID" startValue="1"/>
+        <createSequence sequenceName="SEQ_ANNOUNCEMENT_ID" startValue="1"/>
+    </changeSet>
+
+    <changeSet author="challenge" id="1.0.0-1">
+        <createTable tableName="CHALLENGE">
+            <column name="CHALLENGE_ID" type="BIGINT" autoIncrement="${autoIncrement}" startWith="1">
+                <constraints nullable="false" primaryKey="true" primaryKeyName="PK_CHALLENGE"/>
+            </column>
+            <column name="TITLE" type="VARCHAR(200)">
```
Comment (ahamdi) [inline challenges-services/src/main/resources/db/changelog/challenge-rdbms.db.changelog-1.0.0.xml]: We need to validate the length with @mzorai for the title and the description

File: challenges-services/src/main/resources/db/changelog/challenge-rdbms.db.changelog-1.0.0.xml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    <changeSet author="challenge" id="1.0.0-2">
+        <createTable tableName="ANNOUNCEMENT">
+            <column name="ANNOUNCEMENT_ID" type="BIGINT" autoIncrement="${autoIncrement}" startWith="1">
+                <constraints nullable="false" primaryKey="true" primaryKeyName="PK_ANNOUNCEMENT"/>
+            </column>
+            <column name="CHALLENGE_ID" type="BIGINT">
+                <constraints nullable="false" foreignKeyName="FK_CHALLENGE_ANNOUNCEMENT"
+                             references="CHALLENGE(CHALLENGE_ID)"/>
+            </column>
+            <column name="COMMENT" type="VARCHAR(200)"/>
```
Comment (ahamdi) [inline challenges-services/src/main/resources/db/changelog/challenge-rdbms.db.changelog-1.0.0.xml]: @mzorai Could you precise the maximum length for this ?

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: challenges-services/src/main/resources/db/changelog/challenge-rdbms.db.changelog-1.0.0.xml (hunk 1 above)
Comment (Rdidmohamedelhadi) [reply challenges-services/src/main/resources/db/changelog/challenge-rdbms.db.changelog-1.0.0.xml]: Done !!

**Review by ahamdi — APPROVED** [review]
LGTM

**Pull request merged**